### PR TITLE
Try to robustify Straight Skeleton

### PR DESCRIPTION
**Pull request by sloriot**
Some predicates used in the straight skeleton require several square roots. This robustifies one predicate and increases the precision of 2 constructions used in predicates.

### CODE REVIEW

File: Straight_skeleton_2/include/CGAL/constructions/Straight_skeleton_cons_ftC2.h [hunk 1]
```diff
@@ -35,12 +35,24 @@ Uncertain<Trisegment_collinearity> certified_trisegment_collinearity ( Segment_2
                                                                      , Segment_2<K> const& e1
                                                                      , Segment_2<K> const& e2
                                                                      );
-#ifdef CGAL_USE_CORE  
+#ifdef CGAL_USE_CORE
+
+#ifdef CGAL_USE_GMPXX
+inline CORE::BigFloat to_BigFloat( mpq_class const& n )
```
Comment (mglisse) [inline Straight_skeleton_2/include/CGAL/constructions/Straight_skeleton_cons_ftC2.h]: Shouldn't we add stuff to CORE_coercion_traits.h instead?

**Comment by MaelRL** [comment]
Review status: 0 of 5 files reviewed at latest revision, 2 unresolved discussions.

---

*[Straight_skeleton_2/include/CGAL/Straight_skeleton_2/debug.h, line 274 at r2](https://reviewable.io:443/reviews/cgal/cgal/2917#-L7Ox3f4zOQy2jwg_hBU:-L7Ox3f4zOQy2jwg_hBV:b5m3ow6) ([raw file](https://github.com/cgal/cgal/blob/1132fbe752eac9742dee4c8e5a76a3133c6298a6/Straight_skeleton_2/include/CGAL/Straight_skeleton_2/debug.h#L274)):*
> ```objc
> #ifdef CGAL_STRAIGHT_SKELETON_TRAITS_ENABLE_TRACE
> #  define CGAL_STSKEL_TRAITS_ENABLE_TRACE 1
> #  define CGAL_STSKEL_TRAITS_ENABLE_TRACE_IF(cond) if ((cond))
> ```

Bad change :bug: 

(& test comment)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cgal/cgal/2917)*

**Comment by MaelRL** [comment]
Reviewed 1 of 4 files at r2.
Review status: 1 of 5 files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cgal/cgal/2917)*

**Comment by lrineau** [comment]
@sloriot Do you still want that change in CGAL-4.12-beta2 or CGAL-4.12?

**Comment by lrineau** [comment]
@sloriot Do you still want that change in CGAL-4.12-beta2 or CGAL-4.12?

**Comment by lrineau** [comment]
@sloriot Could you check the status of this PR?

**Comment by sloriot** [comment]
Stalled

**Comment by MaelRL** [comment]
Replaced by #4905.

**Pull request closed without merging**